pkg/controller/apiextensions/requirement/api.go

```diff
@@ -102,8 +111,7 @@ func (a *APIBinder) Bind(ctx context.Context, rq resource.Requirement, cp resour
 // removing the composite resource's requirement reference, and if the composite
 // resource's reclaim policy is "Delete", deleting it.
 func (a *APIBinder) Unbind(ctx context.Context, _ resource.Requirement, cp resource.Composite) error {
-	// TODO(negz): Does this work for unstructured resources?
-	cp.SetRequirementReference(nil)
```

Comment:
I was expecting this to work. Why doesn't it?

Reply:
It sets the ref to the JSON `null`, which is not a valid value according to the schema validation. I believe `*unstructured.Unstructured` has special casing here that interprets `nil` as "delete this field" - we might want to do the same in `pkg/fieldpath`.